chore!: [CMPS-75] Rename root module

corafid opened the pull request:
# Overview

Currently, users who need to use both `cohere` and `compass-sdk` in the same project are facing an issue importing one or other due to clashing top level module name. This PR fixes the issue by renaming `compass-sdk`'s root module (in fact merging `cohere.compass` into `cohere_compass` as the root module.)

# More Info

Python allows multiple packages to own the same top-level model, e.g. cohere. However, it is not as straightforward as, for example, in Java (where you just choose a name like "com.cohere.something" and everything works out of the box), and require some configuration, as well as ensuring the top level module name (in our case `cohere`) doesn't actually have any entity in it, which is the case for `compass-sdk` but not the case for `cohere` Python toolkit.

We have two options:

* Work with the Cohere's SDK team to refactor it  and move everything out of the top-level module name, and then apply the necessary configuration so that these two packages (and potentially future packages) live under the same umbrella top-level module, `cohere`.

* Rename the module name for `compass-sdk` to avoid the clash, e.g. make it cohere_compass.

Both of these would be breaking changes for our customers, but the first one impact many more customers (everyone is using Cohere vs just Compass clients). As such, we will be going with the second option, especially that the second doesn't involve moving of classes/functions/etc to a different scope, so it is just a matter of find/replace.

---

BREAKING CHANGE: Clients need to update their imports of Compass entities from `cohere.compass.<something>` to `cohere_compass.<something>`.

ankush-cohere reviewed (approved):
LGTM

dbaxter-cohere commented on pyproject.toml:
```diff
@@ -1,10 +1,10 @@
 [tool.poetry]
 name = "compass-sdk"
-version = "0.19.0"
+version = "0.21.0"
```
Why are we skipping 0.20.0?

corafid replied:
Good question. Because https://github.com/cohere-ai/cohere-compass-sdk/pull/101/ reserved 0.20.0 :)

dbaxter-cohere replied:
So we should merge them in that order so as to not have the 21 release not contain the 20 contents.

corafid replied:
Yeah, exactly, I am putting this on hold until that PR is merged.

dbaxter-cohere reviewed (commented):
We should follow the semantic rules about breaking changes

```
BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with [MAJOR](http://semver.org/#summary) in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.
```

corafid commented:
> We should follow the semantic rules about breaking changes
> 
> ```
> BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, 
> introduces a breaking API change (correlating with [MAJOR](http://semver.org/#summary) in Semantic Versioning). 
> A BREAKING CHANGE can be part of commits of any type.
> ```

Good suggestion. I updated the title and added a footer as well.
